### ISSUE TITLE
Bug: issue on first build after DD4hep update on system with globaly installed acts

### ISSUE THREAD

**Issue by gagnonlg** [acts-project/acts#1139]
TLDR; I hit an odd issue after upgrading to DD4hep 1.19 on my system today.  I think it's an issue with the DD4hep's cmake setup, but I'm documenting this here until I have time to look into this again.

Longer summary: 

Basically, the build fails with:

```
[ 26%] Generating libActsExamplesDetectorDD4hep.components
ERROR: failed to load libActsExamplesDetectorDD4hep.so: libDDCore.so.1.16: cannot open shared object file: No such file or directory
```
The odd part is that libDDCore.so.1.16 is not present at all on this system. However, as I'm building, there's a previous acts version installed in /usr which is linked against it:
```
 $ ldd /usr/lib/libActsExamplesDetectorDD4hep.so | grep DD
	libActsPluginDD4hep.so => /usr/lib/../lib/libActsPluginDD4hep.so (0x00007f1dfd1b5000)
	libDDCore.so.1.16 => not found
	libDDCore.so.1.16 => not found
```
Building with VERBOSE=1:
```
[ 26%] Generating libActsExamplesDetectorDD4hep.components
cd /home/glg/ACTS/packaging/acts-main/src/acts/build/Examples/Detectors/DD4hepDetector && LD_LIBRARY_PATH=/home/glg/ACTS/packaging/acts-main/src/acts/build/Examples/Detectors/DD4hepDetector:/usr/lib: /usr/bin/listcomponents_dd4hep -o libActsExamplesDetectorDD4hep.components libActsExamplesDetectorDD4hep.so
ERROR: failed to load libActsExamplesDetectorDD4hep.so: libDDCore.so.1.16: cannot open shared object file: No such file or directory
```

But :
```
$ ldd /home/glg/ACTS/packaging/acts-main/src/acts/build/Examples/Detectors/DD4hepDetector/libActsExamplesDetectorDD4hep.so | grep DD
	libActsPluginDD4hep.so => /home/glg/ACTS/packaging/acts-main/src/acts/build/lib/libActsPluginDD4hep.so (0x00007f0bdd889000)
	libDDCore.so.1.19 => /usr/lib/libDDCore.so.1.19 (0x00007f0bdd1cf000)
	libDDParsers.so.1.19 => /usr/lib/libDDParsers.so.1.19 (0x00007f0bdbb06000)
```

So, for some reason, the `listcomponents_dd4hep` ignores the version in the local directory. If I add $PWD/ in front of the .so path, it works as expected. 

Either we're configuring something wrongly, or there's an issue with DD4hep cmake's setup. I propose to keep this open until I can figure that part out. 

Easy workaround: either uninstall the global acts while building the new one, or patch the build script to add the $PWD.

**Comment by paulgessinger** [comment]
I think the step that it's failing at is actually during ROOT dictionary generation. What I think is happening here is that a program executes that tries to look up the shared lib using `LD_LIBRARY_PATH` and nothing from CMake like packages that were found and linked. I'm guessing that's why it's looking at `/usr/lib/` and finding something ther.

Can you try sourcing DD4hep's setup file and compiling again?

**Comment by gagnonlg** [comment]
Well the weird thing here is that LD_LIBRARY_PATH is set to the directory where the freshly-built .so is located when running listcomponents_dd4hep...

Thanks for the suggestion about the setup file, I'll try that next

**Comment by andiwand** [comment]
I imagine this is not relevant anymore. With the spack dependency build this should be much easier now and less error prone.
